### ISSUE TITLE
JCache 1.1 TCK - org.jsr107.tck.event.CacheListenerTest passes, but assertion errors are in the log

### ISSUE THREAD

**Issue by kwart** [hazelcast/hazelcast#12390]
The JCache 1.1 TCK passes for Hazelcast, but the log file contains assertion errors. It seems we experience 2 issues here:

* the TCK handles the result wronly (the test should fail for Hazelcast) ;
* and we have a JCache incompatibility in the CacheListener

https://hazelcast-l337.ci.cloudbees.com/view/JCache/job/JCache-1.1-TCK-branchparam-client-OS/4/console

```
11:59:16 java.lang.AssertionError: Old value should be available for CacheEntryEvent{eventType = REMOVED, isOldValueAvailable = false, value = Lucky, oldValue = null}
11:59:16 	at org.junit.Assert.fail(Assert.java:88)
11:59:16 	at org.junit.Assert.assertTrue(Assert.java:41)
11:59:16 	at org.jsr107.tck.testutil.CacheTestSupport$MyCacheEntryListener.assertOldValueForExpiredRemovedListener(CacheTestSupport.java:204)
11:59:16 	at org.jsr107.tck.testutil.CacheTestSupport$MyCacheEntryListener.onRemoved(CacheTestSupport.java:178)
11:59:16 	at org.jsr107.tck.event.CacheEntryListenerServer.runHandlers(CacheEntryListenerServer.java:151)
11:59:16 	at org.jsr107.tck.event.CacheEntryListenerServer.access$000(CacheEntryListenerServer.java:38)
11:59:16 	at org.jsr107.tck.event.CacheEntryListenerServer$CacheEntryEventOperationHandler.onProcess(CacheEntryListenerServer.java:120)
11:59:16 	at org.jsr107.tck.support.Server$ClientConnection.run(Server.java:310)
```

**Comment by vbekiaris** [comment]
TCK tests do not detect the failures because they assume listeners are executed on the same thread as the cache operations from which the events originate; since in Hazelcast events are published from separate event threads, the assertion failures / exceptions thrown from TCK's test listeners are ignored.

See also https://github.com/jsr107/jsr107tck/issues/141

**Comment by vbekiaris** [comment]
Closing as fix #12385 has been merged and the TCK issue is separately tracked on jsr107 repository.
